### PR TITLE
Fix issue when troop info window isn't shown after moving a single unit between army bars using drag&drop

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
There is a bug that prevents to view troop info after moving a single unit between army bars. How to reproduce:

1. Run the game and load attached savefile;
2. Enter the Burlock castle (Sorceress);
3. Move the single black dragon from Roxana's army (the rightmost stack) to garrison using drag&drop;
4. Try to view troop info for this black dragon using right mouse button. You will not be able to do this.

The same issue is observed when drag&drop is used to move a single unit between meeting heroes. The proposed patch is intended to fix this.

[Army_Troop_Info_Bug.zip](https://github.com/ihhub/fheroes2/files/6446493/Army_Troop_Info_Bug.zip)

### CODE REVIEW

File: src/fheroes2/army/army_bar.cpp [hunk 1]
```diff
@@ -665,3 +642,31 @@ bool ArmyBar::QueueEventProcessing( ArmyBar & bar, std::string * str )
...
+    if ( &troop == selectedTroop ) {
+        return false;
+    }
+
+    // we can redistribute troops either to an empty slot or to a slot containing the same creatures
+    if ( !troop.isValid() || selectedTroop->GetID() == troop.GetID() ) {
+        return true;
+    }
+
+    return false;
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:671]: This code could be written without `if-condition`:
```
return !troop.isValid() || selectedTroop->GetID() == troop.GetID();
```

File: src/fheroes2/army/army_bar.cpp [hunk 2]
```diff
@@ -665,3 +642,31 @@ bool ArmyBar::QueueEventProcessing( ArmyBar & bar, std::string * str )
...
+        return false;
+    }
+
+    // try to redistribute troops if we have a selected troop
+    if ( !isSelected() ) {
+        return false;
+    }
+
+    const ArmyTroop * selectedTroop = GetSelectedItem();
+
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:658]: Could you please remove this empty line? The assertion should be group with variable declaration.

File: src/fheroes2/army/army_bar.cpp [hunk 3]
```diff
@@ -294,6 +293,8 @@ bool ArmyBar::ActionBarCursor( ArmyTroop & troop )
     if ( isSelected() ) {
         const ArmyTroop * troop2 = GetSelectedItem();
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:295]: Could you please remove this empty line? The assertion should be group with variable declaration.

File: src/fheroes2/army/army_bar.cpp [hunk 4]
```diff
@@ -360,14 +361,13 @@ bool ArmyBar::ActionBarLeftMouseSingleClick( ArmyTroop & troop )
 
         ArmyTroop * selectedTroop = GetSelectedItem();
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:363]: Could you please remove this empty line? The assertion should be group with variable declaration.

File: src/fheroes2/army/army_bar.cpp [hunk 5]
```diff
@@ -510,6 +510,8 @@ bool ArmyBar::ActionBarLeftMouseDoubleClick( ArmyTroop & troop )
 
     const ArmyTroop * troop2 = GetSelectedItem();
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:512]: Could you please remove this empty line? The assertion should be group with variable declaration.

File: src/fheroes2/army/army_bar.cpp [hunk 6]
```diff
@@ -600,22 +601,12 @@ bool ArmyBar::ActionBarRightMouseHold( ArmyTroop & troop )
...
-        return false;
-    }
-
-    // try to redistribute troops if we have a selected troop
-    if ( !isSelected() )
-        return false;
-
-    ArmyTroop & selectedTroop = *GetSelectedItem();
+    if ( AbleToRedistributeArmyOnRightMouseSingleClick( troop ) ) {
+        ArmyTroop * selectedTroop = GetSelectedItem();
```
Comment (ihhub) [inline src/fheroes2/army/army_bar.cpp:606]: Could you please remove this empty line? The assertion should be group with variable declaration.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , I left just very minor comments. Would you mind to take a look?

**Comment by ihhub** [comment]
Hi @Branikolog , could you please test this change as well? You know what to test :)

**Comment by Branikolog** [comment]
@ihhub 
Seems OK now... ;)

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you very much for the fix!

**Pull request merged**